### PR TITLE
created functionality for authentication of the user using JWT

### PR DESCRIPTION
**Pull request by dmytrosmetanin**
On the next sprint gonna be slightly updated - gonna add generation of refreshing token.

### CODE REVIEW

File: codeArena/codeArena/settings.py [hunk 1]
```diff
@@ -165,3 +166,15 @@
     'rest_framework_jwt.authentication.JSONWebTokenAuthentication',
   ),
 }
+
+JWT_AUTH = {
+
```
Comment (vvelizariy) [inline codeArena/codeArena/settings.py:171]: Such new lines are not required

File: codeArena/user/serializers.py [hunk 2]
```diff
@@ -1,7 +1,7 @@
 from django.db import models
 from rest_framework import serializers
 from user.models import User, Coder, Role
-from django.contrib.auth.hashers import make_password
+from django.contrib.auth.hashers import get_hasher, make_password
```
Comment (vvelizariy) [inline codeArena/user/serializers.py:4]: Please format import according PEP8, at lease alphabetically

File: codeArena/user/serializers.py [hunk 3]
```diff
@@ -49,6 +52,7 @@ def save(self):
 def create(self, validated_data):
     if 'password' in validated_data:
         validated_data['password'] = make_password(validated_data['password'])
+        print(get_hasher)
```
Comment (vvelizariy) [inline codeArena/user/serializers.py:55]: Seems like forgotten for debug print

File: codeArena/user/views.py [hunk 4]
```diff
@@ -1,11 +1,20 @@
...
 from rest_framework import serializers, status
 from rest_framework.exceptions import ValidationError
 from rest_framework.views import APIView
 from rest_framework.response import Response
-from rest_framework.permissions import AllowAny
-
+from rest_framework.permissions import AllowAny, IsAuthenticated
+from rest_framework.generics import RetrieveUpdateAPIView
+from .serializers import UserSerializer, RoleSerializer
 from .models import Role, User
```
Comment (vvelizariy) [inline codeArena/user/views.py:10]: Format import according PEP8, alpabetically, divide in groups separated by empty line

**Review by OleksandrHavron — APPROVED** [review]

**Review by MetalMari — APPROVED** [review]

**Review by Maksson — APPROVED** [review]

**Pull request merged**